pages/posts/build-oci-image-from-scratch.md

```diff
@@ -0,0 +1,391 @@
+----->  "digest": "sha256:75b148a9a5b61403d082f1930ccc779bfdac5acddb48ed8f90ccdc4219d51268",
+----->  "size": 403
+    },
+    "layers": [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+            "digest": "sha256:c59d5203bc6b8b6ef81f3f6b63e32c28d6e47be806ba8528f8766a4ca506c7ba",
+            "size": 3279768
+        },
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+--------->  "digest": "sha256:c37c06cdec9d6a0f2a2d55deb5aa002b26b37b17c02c2eca908fc062af5f53eb",
+            "size": 1372611
+        }
+    ]
+}
+
+$ mv manifest.json $(sha256sum manifest.json | awk '{print $1}')
+```
+
+Note: the filenames for both config.json and manifest.json would've been the sha256 digests from the first variant of the image we built, I'm choosing to revert them to config.json and manifest.json for presentation purposes.
+
+Since we changed our manifest, we need to update index.json with the new digest and filesize:
+
+```
+$ vim index.json
+{
+    "schemaVersion": 2,
+    "manifests": [
+        {
+            "mediaType": "application/vnd.oci.image.manifest.v1+json",
+--------->  "digest": "sha256:8289bd1bdc2a1fae2d2d717b7c40baaedc4c5c4d9c9f4f1a1b045287067e9f2c",
+--------->  "size": 748,
+            "annotations": {
+                "org.opencontainers.image.ref.name": "hello:alpine1"
+            }
+        }
+    ]
+}
+```
+
+If we pack and load the image again with podman, we can see the modified image in action:
+```
+$ tree
+├── blobs
+│   └── sha256
+│       ├── 75b148a9a5b61403d082f1930ccc779bfdac5acddb48ed8f90ccdc4219d51268
+│       ├── 8289bd1bdc2a1fae2d2d717b7c40baaedc4c5c4d9c9f4f1a1b045287067e9f2c
+│       ├── c37c06cdec9d6a0f2a2d55deb5aa002b26b37b17c02c2eca908fc062af5f53eb
+│       └── c59d5203bc6b8b6ef81f3f6b63e32c28d6e47be806ba8528f8766a4ca506c7ba
+└── index.json
+$ tar -cvf hello-alpine.tar *
+
+$ podman load < hello-alpine.tar
+Getting image source signatures
+Copying blob c59d5203bc6b done   |
+Copying blob c37c06cdec9d done   |
+Copying config 75b148a9a5 done   |
+Writing manifest to image destination
+Loaded image: localhost/hello:alpine
+
+$ podman run --rm localhost/hello:alpine reader
+Hello, reader!
+real    0m 0.00s
+user    0m 0.00s
+sys     0m 0.00s
+
+$ podman image ls hello
+REPOSITORY   TAG       IMAGE ID       CREATED   SIZE
+hello        scratch   25e8b3bd9720   N/A       3.67MB
+hello        alpine    3d0268e9a91e   N/A       11MB
+```
+
+So there we have it, our two images loaded into the docker daemon. We can run both of them and see the same output from both of them. One interesting thing to note is the size difference between the two images. Alpine, while still having a lower footprint is larger than scratch, but nothing extraordinary. Hopefully this example demonstrated interaction between different layers in an image and how a manifest file can represent multiple images from a single file.
```

Comment:
> One interesting thing to note is the size difference between the two images. Alpine, while still having a lower footprint is larger than scratch, but nothing extraordinary.

Why is it interesting? And why do you have to mention it at all?

> Hopefully this example demonstrated interaction between different layers in an image and how a manifest file can represent multiple images from a single file.

Drop this sentence. If you have to explain in the end what you were trying to achieve, you failed to achieve it. :smile: